### ISSUE TITLE
Nginx/FPM timeout

### ISSUE THREAD

**Issue by otravers** [devgeniem/docker-wordpress-development#1]
I started working with your image a few days ago and the following errors started to happen today with my local WordPress project set up with **devgeniem/wordpress-development-server:latest**

`[error] 311#311: *1 upstream timed out (110: Operation timed out) while reading upstream, client: 172.18.0.1, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php-fpm.sock:"
`

I see that the underlying **docker-wordpress** project had several commits in the past 24 hours, so maybe it's the cause of my issue?

If I pull devgeniem/wordpress-server:latest, I have no timeouts, but if I use devgeniem/wordpress--development-server:latest, the container times out very quickly. I need Xdebug!

PS; your project and container layout is very interesting, thank you for open sourcing it. May I suggest you name your Docker Hub and Github projects more consistently? It took me a while to wrap my head around all the images and source bases.

**Comment by onnimonni** [comment]
Thanks for reporting this.

I haven't done any changes that I'm aware to this image. Underlying project has new commits because we are preparing a version with nginx pagespeed and openresty modules, but I didn't mean to change anything in the `php5.6` and `php7.0` images.

Have you tried to restart or recreate your container?

**Comment by otravers** [comment]
I have deleted and recreated the container several times: development always times out, while underlying docker-wordpress works great. I just went back and forth 3 times between the two images.

PS: my setup doesn't use Redis, I've disabled it several days ago.

**Comment by onnimonni** [comment]
Does this happen only for larger queries?

What happens if you visit something small eg. `/robots.txt` or `/wp-login.php`?

I'm thinking if the fastcgi timeout might be too small for your setup.
You can try to change this env: `NGINX_FASTCGI_TIMEOUT=300`

**Comment by otravers** [comment]
I can load, say a page with phpinfo(), though it's slow. I changed the ENV NGINX_FASTCGI_TIMEOUT to 300 in my Docker Compose, now the site's home page still times out eventually. The exact same page loads almost instantly with docker-wordpress.

Edit: it appears the problem come to a setting here:
https://github.com/devgeniem/docker-wordpress-development/blob/master/php7.0/etc/php7/php-fpm.ini

I cloned the development project, commented out "COPY etc/ /etc/" in the Dockerfile, loaded that image in my project, and pages load instantly. I'll see whether I can track down the root cause.

New edit: I commented things in and out in etc/php-fpm.ini, just enabling xdebug.remote_enable leads to the timeouts. I investigated further and it turns out the XDEBUG_REMOTE_HOST value I was putting in Dockerfile is the reason why. Sorry for the trouble, I've been struggling to get this to work with PhpStorm and this sent me down the rabbit hole.

**Comment by onnimonni** [comment]
After you comment out `COPY etc/ /etc/` 2 things happen:
- the production image is allowed to use opcache: https://github.com/devgeniem/docker-wordpress/blob/master/php7.0/rootfs/etc/php7/php-fpm.ini#L55
- Production image does not use xdebug which slows development image quite a bit

If you use docker for mac there's also problem with slow docker mounts which make it unbearable to use.

**Comment by otravers** [comment]
Yes, I looked into the guts of /etc and the differences between this image and the underlying one. I am including /etc, I was just playing around to try and pinpoint the source of these timeouts.

The issue I had reported earlier was of my own making because I was not using the correct IP address for xdebug.remote_host. I found that by checking xdebug.log which could not connect to localhost, 127.0.0.1, or 10.0.75.1. I fixed it by using (192.168.1.2, the host's IP) and Xdebug was successfully connecting for a couple of days.

However, this stopped working today. I still have the correct IP in my setup and now I see timeouts in xdebug.log. I see that docker-wordpress got some new commits today, though I checked and that image still works great for me. So it's really an issue with Xdebug.

I'm using Docker for Windows, mounts seem to work OK there.
